### PR TITLE
Updated to follow tfsec best practices AWS016

### PR DESCRIPTION
**Pull request by ghost**
https://tfsec.dev/docs/aws/AWS016/

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -18,6 +18,25 @@ data "archive_file" "lambda" {
...
 
+
+##########################################################################
+#                                                                        #
+#                               KMS Key                                  #
+#                                                                        #
+##########################################################################
+
+resource "aws_kms_key" "marbot" {
+  description         = "KMS CMK for ${var.module_name}"
```
Comment (michaelwittig) [inline main.tf]: why do you think alarms should use an encrypted topic? they don't contain PII data. But we add $1 / month to the bill.

File: main.tf [hunk 2]
```diff
@@ -572,8 +592,9 @@ resource "aws_iam_role_policy" "cloud_watch_alarm_filter" {
 }
 
 resource "aws_lambda_function" "cloud_watch_alarm_filter" {
+  depends_on = [data.archive_file.lambda]
```
Comment (michaelwittig) [inline main.tf]: can you explain why this is needed?

File: variables.tf [hunk 3]
```diff
@@ -1,3 +1,9 @@
+variable "module_name" {
```
Comment (michaelwittig) [inline variables.tf:1]: can you explain this new var?

**Review by michaelwittig — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (worwoods) [reply main.tf]: @michaelwittig tfsec is flagging this as an error and the flag can be cleared by either ignoring tfsec (not an option in my case), inserting an exception into the code acceptable or adding the encryption

I come from a background where the default behaviour is to add the layer if all things are equal, which in this case I believe them to be. I appreciate all the dollars add up and this is undesirable for sum and happy to put in the ignore statement if you'd prefer.

Comment (michaelwittig) [reply main.tf]: What about a flag to turn encryption on (off by default)?

File: main.tf [hunk 4]
```diff
@@ -14,10 +14,29 @@ data "aws_region" "current" {}
 
 data "archive_file" "lambda" {
   type        = "zip"
-  output_path = "${path.module}/lambda.zip"
+  output_path = "./lambda.zip"
```
Comment (michaelwittig) [inline main.tf:17]: can you explain this change?

File: main.tf (hunk 1 above)
Comment (worwoods) [reply main.tf]: will take a look later today

Comment (worwoods) [reply main.tf]: Removed due to time constraints

File: main.tf (hunk 2 above)
Comment (worwoods) [reply main.tf]: I was having a problem with archive files being created, turns out they now get created in plan not apply and wasn't making in between CI pipeline stages, also why I have now set to be created in the root of the terraform that calls the module.

File: main.tf (hunk 4 above)
Comment (worwoods) [reply main.tf:17]: During my testing I found an issue related to the fact that `apply` know longer creates the archive but instead `plan` when used in a pipeline this was seen to be a major problem as some tools have a limit on how filesare allowed to traverse stages. Moving the file to this location simplifies the troubleshooting of the issue and allows consistent behaviour. I'm certainly not an expert when it comes to lambdas but this is a common pattern if not best practive.

**Pull request closed without merging**